src/common/src/mlib/duration.h

```diff
@@ -163,35 +160,45 @@ _mlibDurationCopy (mlib_duration d)
 }
 
 // Duration scalar multiply
-#define _mlibDurationInfixOperator_mul(LHS, Fac) _mlibDurationMultiply (_mlibDurationArgument (LHS), (Fac))
+#define _mlibDurationInfixOperator_mul(LHS, Fac) \
+   _mlibDurationMultiply (_mlibDurationArgument (LHS), mlib_upsize_integer (Fac))
 static inline mlib_duration
-_mlibDurationMultiply (const mlib_duration dur, int fac) mlib_noexcept
+_mlibDurationMultiply (const mlib_duration dur, mlib_upsized_integer fac) mlib_noexcept
 {
    mlib_duration ret = {0};
-   if (mlib_mul (&ret._rep, dur._rep, fac)) {
-      if ((dur._rep < 0) != (fac < 0)) {
+   uintmax_t bits;
+   if ((mlib_mul) (&bits, true, true, (uintmax_t) dur._rep, fac.is_signed, fac.bits.as_unsigned)) {
+      if ((dur._rep < 0) != (fac.is_signed && fac.bits.as_signed < 0)) {
```

Comment:
Perhaps there should be a three-way `mlib_compare()` and/or an `mlib_is_negative()`. (OK with deferring to another PR.)

> `if ((dur._rep < 0) != (fac.is_signed && fac.bits.as_signed < 0)) {`

```c
if ((dur._rep < 0) != mlib_is_negative(fac)) {
```

> `if ((div.is_signed && div.bits.as_signed == -1) && ...) {`

```c
if (mlib_compare(div, -1) == 0 && ...) {
```

Reply:
I'd like to have `mlib_upsize_integer` support passing an already-upsized integer object, allowing it to be used with all the safe-int functions. It would make using upsized-integers a lot more useful outside of the generic contexts, but that would require `_Generic`. I'll defer any changes for now.

(Also "upsized integer" is a bad name. May rename to "wide integer" or something.)